modules/tasks/tasks/src/test/java/org/motechproject/tasks/service/impl/TaskServiceImplTest.java

```diff
@@ -226,6 +242,8 @@ public void shouldSaveTaskWithEmptyActionInputFields() {
 
         taskService.save(task);
         verify(triggerHandler).registerHandlerFor(task.getTrigger().getEffectiveListenerSubject());
+        // When task has not set number of retries, it should not register handler for retries
+        verifyNoMoreInteractions(triggerHandler);
```

Comment:
this won't be executed since taskService.save throws exception. Can you refactor and put those checks in the finally block?